internal/ethapi/api.go

```diff
@@ -1729,13 +1757,27 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, ha
 	// Derive the sender.
 	bigblock := new(big.Int).SetUint64(blockNumber)
 	signer := types.MakeSigner(s.b.ChainConfig(), bigblock)
+	return marshalReceipt(ctx, s.b, receipt, blockHash, blockNumber, signer, tx, int(index)), nil
```

Comment:
so it always return nil here for err ? 
In the previous logic 
```
		header, err := s.b.HeaderByHash(ctx, blockHash)
		if err != nil {
			return nil, err
		}
```
I still see it returns err when can't get HeaderbyHash

Reply:
The blockHash is the result of this call: https://github.com/axieinfinity/ronin/pull/641/files#diff-c426ecd2f7d247753b9ea8c1cc003f21fa412661c1f967d203d4edf8163da344R1744. So when reaching here, it's very unlikely we get the error. As you can see at that line, when that transaction does not exist or some error happens, it still returns nil in the error.

Reply:
oke,  so expect the logic will be same with the above logic